packages/sdk/src/application/ApplicationClient.ts

```diff
@@ -0,0 +1,411 @@
+
+      await this._client.delete(queryString, {});
+      return Promise.resolve();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async listConversations(
+    props: ListConversationsRequest,
+  ): Promise<ListConversationsResponse> {
+    const { ownerId, appId, pageSize, pageToken } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/v1alpha/namespaces/${ownerId}/apps/${appId}/conversations`,
+        pageSize,
+        pageToken,
+      });
+
+      const data = await this._client.get<ListConversationsResponse>(
+        queryString,
+        {},
+      );
+
+      return Promise.resolve(data);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async createConversation(
+    props: CreateConversationRequest,
+  ): Promise<CreateConversationResponse> {
+    const { ownerId, appId, payload } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/v1alpha/namespaces/${ownerId}/apps/${appId}/conversations`,
+      });
+
+      const response = await this._client.post<CreateConversationResponse>(
+        queryString,
+        {
+          body: JSON.stringify(payload),
+        },
+      );
+      return Promise.resolve(response);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async updateConversation(
+    props: UpdateConversationRequest,
+  ): Promise<UpdateConversationResponse> {
+    const { namespaceId, appId, conversationId, payload } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/v1alpha/namespaces/${namespaceId}/apps/${appId}/conversations/${conversationId}`,
+      });
+
+      const response = await this._client.put<UpdateConversationResponse>(
+        queryString,
+        {
+          body: JSON.stringify(payload),
+        },
+      );
+      return Promise.resolve(response);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async deleteConversation(props: DeleteConversationRequest): Promise<void> {
+    const { ownerId, appId, conversationId } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/v1alpha/namespaces/${ownerId}/apps/${appId}/conversations/${conversationId}`,
+      });
+
+      await this._client.delete(queryString, {});
+      return Promise.resolve();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async listMessages(
+    props: ListMessagesRequest,
+  ): Promise<ListMessagesResponse> {
+    const { ownerId, appId, conversationId } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/v1alpha/namespaces/${ownerId}/apps/${appId}/conversations/${conversationId}/messages`,
+      });
+
+      const response = await this._client.get<ListMessagesResponse>(
+        queryString,
+        {},
+      );
+      return Promise.resolve(response);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
```

Comment:
@thewbuk listMessages support pageToken and pageSize in the API, so this endpoint should be like all the others list sdk functions, support pagination